# .yaml config file
#
# The PID gains and controller settings must be saved in a yaml file that gets loaded
# to the param server via the roslaunch file (astra_rover_control.launch).

astra_rover:
  # Publish all joint states -----------------------------------
  # Creates the /joint_states topic necessary in ROS
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Wheel Velocity Controllers ---------------------------------------
  wheel_left_1_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_left_1_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}
  wheel_left_2_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_left_2_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}
  wheel_left_3_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_left_3_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}

  wheel_right_1_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_right_1_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}
  wheel_right_2_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_right_2_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}
  wheel_right_3_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_right_3_joint
    pid: {p: 10.0, i: 0.0, d: 0.0}

  # Wheel Trees Position Controllers ---------------------------------------
  rocker_hinge_left_position_controller:
    type: effort_controllers/JointPositionController
    joint: rocker_hinge_left
    pid: {p: 2200.0, i: 10.0, d: 10.0}
  bogey_hinge_left_position_controller:
    type: effort_controllers/JointPositionController
    joint: bogey_hinge_left
    pid: {p: 4200.0, i: 10.0, d: 10.0}

  rocker_hinge_right_position_controller:
    type: effort_controllers/JointPositionController
    joint: rocker_hinge_right
    pid: {p: 4200.0, i: 10.0, d: 10.0}
  bogey_hinge_right_position_controller:
    type: effort_controllers/JointPositionController
    joint: bogey_hinge_right
    pid: {p: 2200.0, i: 10.0, d: 10.0}


  # Wheel Steering Position Controllers ---------------------------------------
  bogey_steering_left_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: bogey_steering_left_joint
    pid: {p: 1000.0, i: 10.0, d: 100.0}
  rocker_steering_left_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: rocker_steering_left_joint
    pid: {p: 1000.0, i: 10.0, d: 100.0}

  bogey_steering_right_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: bogey_steering_right_joint
    pid: {p: 1000.0, i: 10.0, d: 100.0}
  rocker_steering_right_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: rocker_steering_right_joint
    pid: {p: 1000.0, i: 10.0, d: 100.0}






